expression/builtin_cast.go

```diff
@@ -1907,6 +1908,33 @@ func WrapWithCastAsDecimal(ctx sessionctx.Context, expr Expression) Expression {
 	return BuildCastFunction(ctx, expr, tp)
 }
 
+// charsetConvertMap contains the builtin functions which arguments need to be converted to the correct charset.
+var charsetConvertMap = map[string]struct{}{
+	ast.Hex: {}, ast.Length: {}, ast.OctetLength: {}, ast.ASCII: {},
```

Comment:
Maybe other functions can also be added here, like md5(), sha1()... 

reset, LGTM

Reply:
Yea. We need to check them one by one later.